pkg/issuer/acme/dns/route53/route53.go

```diff
@@ -98,9 +97,14 @@ func (d *sessionProvider) GetSession(ctx context.Context) (aws.Config, error) {
 		return aws.Config{}, fmt.Errorf("unable to create aws config: %s", err)
 	}
 
+	// Explicitly set the region to aws-global so that AssumeRole can be used
+	// with the global sts endpoint.
+	stsCfg := cfg.Copy()
+	stsCfg.Region = "aws-global"
```

Comment:
I wanted some evidence  for the statement in the release-note, that cert-manager 1.14 behaviour was to always use the global endpoints.

This entry from the aws-sdk-v1 changelog, kind of supports that idea.

https://github.com/aws/aws-sdk-go/blob/f2d0da833d2ce3c72cc5c5d2273ca2fba16926f3/CHANGELOG.md#L14772-L14793

> ### SDK Enhancements
> * `aws/endpoints`: Adds support for STS Regional Flags ([#2779](https://github.com/aws/aws-sdk-go/pull/2779))
>   * Implements STS regional flag, with support for `legacy` and `regional` options. Defaults to `legacy`. Legacy, will force all regions specified in aws/endpoints/sts_legacy_regions.go to resolve to the STS global endpoint, sts.amazonaws.com. This is the SDK's current behavior.
>   * When the flag's value is `regional` the SDK will resolve the endpoint based on the endpoints.json model. This allows STS to update their service's modeled endpoints to be regionalized for all regions. When `regional` turned on use `aws-global` as the region to use the global endpoint.
>   * `AWS_STS_REGIONAL_ENDPOINTS=regional` for environment, or `sts_regional_endpoints=regional` in shared config file.
>   * The regions the SDK defaults to the STS global endpoint in `legacy` mode are: 
>     * ap-northeast-1
>     * ap-south-1
>     * ap-southeast-1
>     * ap-southeast-2
>     * aws-global
>     * ca-central-1
>     * eu-central-1
>     * eu-north-1
>     * eu-west-1
>     * eu-west-2
>     * eu-west-3
>     * sa-east-1
>     * us-east-1
>     * us-east-2
>     * us-west-1
>     * us-west-2

Reply:
But then I look at the original code in https://github.com/cert-manager/cert-manager/pull/6671 and I see the following comment:
> // If ambient credentials aren't permitted, always set the region, even if to
> // empty string, to avoid it falling back on the environment.
> // this has to be set after session is constructed

This makes me think that `aws-global` should be the region we use, if the user omitted the region and if `! useAmbientCredentials`. 

I think that comment (which still exists) needs updating.
And a comment here or somewhere explaining that we force the use of global STS endpoints, for backwards compatibility reasons.

Reply:
I'll expand the comments. However, note that the comment in 6671 is talking about the region for route53.